opm/material/components/SimpleHuDuanH2O.hpp

```diff
@@ -0,0 +1,399 @@
+     */
+    template <class Evaluation>
+    static Evaluation gasViscosity(const Evaluation& /*temperature*/,
+                                   const Evaluation& /*pressure*/)
+    {
+        return 1e-05;
+    }
+
+    /*!
+     * \brief The dynamic viscosity \f$\mathrm{[Pa*s]}\f$ of pure water.
+     *
+     * \param temperature temperature of component in \f$\mathrm{[K]}\f$
+     * \param pressure pressure of component in \f$\mathrm{[Pa]}\f$
+     */
+    template <class Evaluation>
+    static Evaluation liquidViscosity(const Evaluation& temperature, const Evaluation& pressure)
+    {
+        if (temperature > 570) {
+            std::ostringstream oss;
+            oss << "Viscosity of water based on Hu et al is too different from IAPWS for T above 570K and "
+                << "(T = " << temperature << ")";
+            throw NumericalIssue(oss.str());
+        }
+
+        const Evaluation& rho = liquidDensity(temperature, pressure);
+        return Common::viscosity(temperature, rho);
+    }
+
+private:
+
+    /*!
+     * \brief The density of pure water in \f$\mathrm{[kg/m3]}\f$
+     *
+     * \param temperature temperature of component in \f$\mathrm{[K]}\f$
+     * \param pressure pressure of component in \f$\mathrm{[Pa]}\f$
+     */
+    template <class Evaluation>
+    static Evaluation liquidDensity_(const Evaluation& T, const Evaluation& pressure) {
+        // Hu, Duan, Zhu and Chou: PVTx properties of the CO2-H2O and CO2-H2O-NaCl
+        // systems below 647 K: Assessment of experimental data and
+        // thermodynamics models, Chemical Geology, 2007.
+        if (T > 647 || pressure > 100e6) {
+            std::ostringstream oss;
+            oss << "Density of water is only implemented for temperatures below 647K and "
+                << "pressures below 100MPa. (T = " << T << ", p=" << pressure;
+            throw NumericalIssue(oss.str());
+        }
+
+        Evaluation p = pressure / 1e6; // to MPa
+        Scalar Mw = molarMass() * 1e3; //kg/kmol
+
+        static const Scalar k0[5] = { 3.27225e-07, -4.20950e-04, 2.32594e-01, -4.16920e+01, 5.71292e+03 };
+        static const Scalar k1[5] = { -2.32306e-10, 2.91138e-07, -1.49662e-04, 3.59860e-02, -3.55071 };
+        static const Scalar k2[3] = { 2.57241e-14, -1.24336e-11, 5.42707e-07 };
+        static const Scalar k3[3] = { -4.42028e-18, 2.10007e-15, -8.11491e-11 };
+        Evaluation k0_eval = 1e-3 * (((k0[0]*T + k0[1])*T + k0[2])*T + k0[3] + k0[4]/T);
+        Evaluation k1_eval = 1e-2 * (((k1[0]*T + k1[1])*T + k1[2])*T + k1[3] + k1[4]/T);
+        Evaluation k2_eval = 1e-1 * ((k2[0]*T + k2[1])*T*T + k2[2]);
+        Evaluation k3_eval = (k3[0]*T + k3[1])*T*T + k3[2];
+
+        // molar volum (m³/kmol):
+        Evaluation vw = ((k3_eval*p + k2_eval)*p + k1_eval)*p + k0_eval;
+
+        // density kg/m3
+        return Mw / vw;
+
+    }
+
+};
+
+template <class Scalar>
+const Scalar SimpleHuDuanH2O<Scalar>::R = Opm::Constants<Scalar>::R / 18e-3;
```

Comment:
can't this just be static constexpr and given the value inline ?

Reply:
I guess so. But this is the way it's done for the other water components. I prefer not to change it in this file only.